R/estimate_contrasts.R

```diff
@@ -11,9 +11,32 @@
 #'   "bonferroni", "BH", "BY", "fdr" or "none". See the p-value adjustment
 #'   section in the `emmeans::test` documentation.
 #' @param adjust Deprecated in favour of `p_adjust`.
+#' @param effectsize Desired measure of standardized effect size, one of "none"
+#' (default), "emmeans", "bootES".
+#' @param bootES_type Specifies the type of effect-size measure to
+#' estimate when using `effectsize = "bootES"`. One of `c("unstandardized",
+#' "cohens.d", "hedges.g", "cohens.d.sigma", "r", "akp.robust.d")`. See`
+#' effect.type` argument of [bootES::bootES] for details.
+#' @param bootstraps The number of bootstrap resamples to perform.
 #'
 #' @inherit estimate_slopes details
 #'
+#' @section Effect Size: By default, `estimate_contrasts` reports no standardized effect size
+#' on purpose. Should one request one, some things are to keep in mind. As the
+#' authors of `emmeans` write, "There is substantial disagreement among
+#' practitioners on what is the appropriate sigma to use in computing effect
+#' sizes; or, indeed, whether any effect-size measure is appropriate for some
+#' situations. The user is completely responsible for specifying appropriate
+#' parameters (or for failing to do so)."
+#'
+#' In particular, effect size methods `"emmeans"` and `"bootES"` do not correct
+#' for covariates in the model, so should probably only be used when there is
```

Comment:
I don't know about bootES, but emmeans does return adjusted (corrected?) effect sizes.

Reply:
Thanks, I removed `emmeans` from that paragraph and only left `bootES`.

Reply:
Actually the reason why I included `emmeans` in this paragraph is that in #easystats/effectsize351, @bwiernik writes:

> Method 2: Standardize using the MSE (sigma). This is what `emmeans::eff_size()` does. This works when the contrasts are the only predictors in the model, but not when there are covariates. The response variance accounted for by the covariates should not be removed from the SD used to standardize. Otherwise, d will be overestimated as shown above. I can pull some references from Larry Hedges about this.

So I'm not sure I understand correctly whether `emmeans` correct (correctly) for covariates or not?

Reply:
Well that depends on what you think the correct correction is 😅
emmeans uses the conditional SD, whereas Brenton is in the opinion that the marginal SD should be used (: